core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -427,5 +433,363 @@ namespace ledger {
+                baseReq.AddMember(cosmos::constants::kFrom, buffer, allocator);
+
+                buffer = makeStringValue(memo_, allocator);
+                baseReq.AddMember(cosmos::constants::kMemo, buffer, allocator);
+
+                buffer = makeStringValue(chainId_, allocator);
+                baseReq.AddMember(cosmos::constants::kChainId, buffer, allocator);
+
+                buffer = makeStringValue(accountNumber_, allocator);
+                baseReq.AddMember(cosmos::constants::kAccountNumber, buffer, allocator);
+
+                buffer = makeStringValue(accountSequence_, allocator);
+                baseReq.AddMember(cosmos::constants::kSequence, buffer, allocator);
+
+                buffer = makeStringValue(gas_, allocator);
+                baseReq.AddMember(cosmos::constants::kGas, buffer, allocator);
+
+                auto fees = makeAmountArray(fees_.amount, allocator);
+                baseReq.AddMember(cosmos::constants::kFees, fees, allocator);
+
+                auto simulate = rapidjson::Value(rapidjson::kTrueType);
+                baseReq.AddMember(cosmos::constants::kSimulate, simulate, allocator);
+
+                return baseReq;
+            }
+
+        private:
+            std::string from_;
+            std::string memo_;
+            std::string chainId_;
+            std::string accountNumber_;
+            std::string accountSequence_;
+            std::string gas_;
+            std::string gasAdjustment_;
+            cosmos::Fee fees_;
+            bool simulate_;
+        };
+
+        rapidjson::Value makeBaseReq(const std::shared_ptr<api::CosmosLikeTransaction>& transaction,
+                                     const std::shared_ptr<api::CosmosLikeMessage>& message,
+                                     rapidjson::Document::AllocatorType& allocator)
+        {
+            const auto tx = std::dynamic_pointer_cast<CosmosLikeTransactionApi>(transaction);
+            const auto msg = std::dynamic_pointer_cast<CosmosLikeMessage>(message);
+            const auto baseReq = BaseReq(
+                    msg->getFromAddress(),
+                    tx->getMemo(),
+                    networks::getCosmosLikeNetworkParameters(tx->getCurrency().name).ChainId,
+                    tx->getAccountNumber(),
+                    tx->getAccountSequence(),
+                    tx->getTxData().fee.gas.toString(),
+                    std::string("1"),
+                    tx->getTxData().fee,
+                    true);
+
+            return baseReq.toJson(allocator);
+        }
+
+        struct JsonObject
+        {
+            rapidjson::GenericStringRef<char> name;
+            rapidjson::Value& value;
+
+            JsonObject(const char* name, rapidjson::Value& value)
+                : name(rapidjson::StringRef(name)), value(value)
+            {}
+        };
+
+        template <typename... JsonObj>
+        std::string makeJsonFrom(rapidjson::Document& document,
```

Comment:
There's an issue on my side with this method, the resulting output looks like `"{\"base_req\":null,\"delegator_address\":{\"amount\":\"1000000\",\"denom\":\"uatom\"},\"validator_address\":null,\"amount\":null}"` for example.

I can't understand why and if it's something on my side. We can always merge and test on the main branch...